src/Algebra/Graph/NonEmpty.hs

```diff
@@ -274,7 +287,9 @@ vertices1 = overlays1 . fmap vertex
 -- 'edgeCount' . edges1   == 'Data.List.NonEmpty.length' . 'Data.List.NonEmpty.nub'
 -- @
 edges1 :: NonEmpty (a, a) -> NonEmptyGraph a
-edges1 = overlays1 . fmap (uncurry edge)
+edges1 (x :| xs) = foldr (Overlay . edgeTuple) (edgeTuple x) xs
+  where
+    edgeTuple = uncurry edge
```

Comment:
I would drop `edgeTuple` -- I don't think duplicating `uncurry edge` is such a big deal.